### PR TITLE
Non-Kotlin modules are excluded from checking that the module contains detekt

### LINKED ISSUE #81: The Detekt plugin throws an exception in the proxy module
If the root of the gradle infrastructure is different from the root of gradle, the detekt plugin treats gradle proxy modules as subprojects and throws an exception `The "feature" modules do not contain "detekt" or the "redmadrobot.detekt" plugin.` One solution is to check that the subproject contains a build file.

### PR DESCRIPTION
**Pull request by rwqwr**
Closes #81

### CODE REVIEW

**Comment by osipxd** [comment]
Nice!
It would be better to make functions relocation and fixes in separated commits to make it easier to see what was changed.

File: infrastructure/src/main/kotlin/detekt/DetektConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+import io.gitlab.arturbosch.detekt.Detekt
+import io.gitlab.arturbosch.detekt.DetektCreateBaselineTask
+import org.gradle.api.Project
+import org.gradle.api.tasks.SourceTask
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.kotlin.dsl.getByType
+import org.gradle.kotlin.dsl.named
+import org.gradle.kotlin.dsl.register
+
+internal class DetektConfiguration(
```
Comment (osipxd) [inline infrastructure/src/main/kotlin/detekt/DetektConfiguration.kt]: Is it better than keeping all functions in `DetektPlugin`?
Plugin class now is nearly empty and all logic is still in one place here.

File: infrastructure/src/main/kotlin/DetektPlugin.kt [hunk 2]
```diff
@@ -71,27 +61,32 @@ private fun Project.configureDetektAllTasks(extension: RedmadrobotExtension, inf
...
+        "detekt${DetektPlugin.BASELINE_KEYWORD}All",
+    ) {
         description = "Runs over whole code base without the starting overhead for each module."
     }
 
     if (project.isInfrastructureRootProject) {
-        val variantRegex = Regex("detekt($BASELINE_KEYWORD)?([A-Za-z]+)All$")
-        val taskRegex = Regex("^(${project.path.orEmpty()}:)?$variantRegex")
+        val variantRegex = Regex("detekt(${DetektPlugin.BASELINE_KEYWORD})?([A-Za-z]+)All$")
+        val taskRegex = Regex("^(:?${project.name.orEmpty()}:)?$variantRegex")
```
Comment (osipxd) [inline infrastructure/src/main/kotlin/DetektPlugin.kt]: Will project.name include parent projects also? For example, we have a hierarchy:
```
libs <-- parent project
libs.lib1 <-- project where detekt plugin applied
```
Will `project.name` return `libs:lib1` or `lib1`?

File: infrastructure/src/main/kotlin/detekt/DetektConfiguration.kt (hunk 1 above)
Comment (rwqwr) [reply infrastructure/src/main/kotlin/detekt/DetektConfiguration.kt]: I thought that the Detekt configuration project had become too complex, and Detekt reported a "TooManyFunctions" error. As I recall, we discussed this. But I also like to keep everything in one place. Moved the logic back.

File: infrastructure/src/main/kotlin/DetektPlugin.kt (hunk 2 above)
Comment (rwqwr) [reply infrastructure/src/main/kotlin/DetektPlugin.kt]: Good point! No, that will be fixed.  Hadn't thought of that.

**Pull request merged**